crates/core/src/network/connection_manager.rs

```diff
@@ -271,6 +304,20 @@ impl<T: TransportConnect> ConnectionManager<T> {
         if let Some(connection) = maybe_connection {
             return Ok(connection);
         }
+
+        let is_gone = {
+            self.inner
+                .lock()
+                .observed_generations
+                .get(&node_id.as_plain())
+                .map(|status| status.generation <= node_id.generation() && status.gone)
```

Comment:
Shouldn't we reject connection attempts if the `node_id.generation` is smaller than `status.generation` or equal and `status.gone`? If `node_id.generation > status.generation`, then it's probably a new node about which we can't say anything yet.

Reply:
That's right. I guess I need to get some sleep :(